samples/net/cloud/google_iot_mqtt/prj.conf

```diff
@@ -6,6 +6,8 @@ CONFIG_NET_TCP=y
 CONFIG_NET_DHCPV4=y
 CONFIG_STDOUT_CONSOLE=y
 
+CONFIG_LEGACY_TIMEOUT_API=y
```

Comment:
I do not think we should re-enable the legacy timeout as we are trying to get rid of it. I created a separate PR #25122 that converts the google iot sample to new timeouts so we do not need to apply this specific commit. The rest of the PR looks fine.